flink-filesystems/flink-azure-fs-hadoop/src/test/java/org/apache/flink/fs/azurefs/AzureFileSystemBehaviorITCase.java

```diff
@@ -70,12 +60,30 @@ public class AzureFileSystemBehaviorITCase extends FileSystemBehaviorTestSuite {
 
     private static final String TEST_DATA_DIR = "tests-" + UUID.randomUUID();
 
-    // Azure Blob Storage defaults to https only storage accounts. We check if http support has been
-    // enabled on a best effort basis and test http if so.
-    @Parameterized.Parameters(name = "Scheme = {0}")
-    public static List<String> parameters() throws IOException {
-        boolean httpsOnly = isHttpsTrafficOnly();
-        return httpsOnly ? Arrays.asList("wasbs") : Arrays.asList("wasb", "wasbs");
+    /**
+     * Azure Blob Storage defaults to https only storage accounts, tested in the base class.
+     *
+     * <p>This nested class repeats the tests with http support, but only if a best effort check on
+     * https support succeeds.
+     */
+    static class HttpSupportAzureFileSystemBehaviorITCase extends AzureFileSystemBehaviorITCase {
```

Comment:
hm, I'm wondering whether it should be a separate test class instead of an inner one. But I don't have a strong opinion here. :thinking: So, I guess, it's good enough :shrug:

Reply:
I don't think this makes a significant difference to warrant the change!